pkg/operator/k8sutil/prometheus.go

```diff
@@ -0,0 +1,106 @@
+package k8sutil
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	monitoringclient "github.com/coreos/prometheus-operator/pkg/client/versioned"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sYAML "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func getMonitoringClient() (*monitoringclient.Clientset, error) {
+	cfg, err := clientcmd.BuildConfigFromFlags("", "")
+	client, err := monitoringclient.NewForConfig(cfg)
+	if err != nil {
+		err = fmt.Errorf("failed to get monitoring client. %+v", err)
+	}
+	return client, err
+}
+
+// GetServiceMonitorDefinition returns a prometheus servicemonitor object
+func GetServiceMonitorDefinition(name, namespace string) *monitoringv1.ServiceMonitor {
+	serviceMonitorDefinition := monitoringv1.ServiceMonitor{}
+	serviceMonitorDefinition.SetName(name)
+	serviceMonitorDefinition.SetNamespace(namespace)
+	return &serviceMonitorDefinition
+}
+
+// CreateOrUpdateServiceMonitor creates serviceMonitor object or an error
+func CreateOrUpdateServiceMonitor(serviceMonitorDefinition *monitoringv1.ServiceMonitor) (*monitoringv1.ServiceMonitor, error) {
+	name := serviceMonitorDefinition.GetName()
+	namespace := serviceMonitorDefinition.GetNamespace()
+	logger.Debugf("creating servicemonitor %s", name)
+	client, err := getMonitoringClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get monitoring client. %+v", err)
+	}
+	sm, err := client.MonitoringV1().ServiceMonitors(namespace).Create(serviceMonitorDefinition)
+	if err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return nil, fmt.Errorf("failed to create servicemonitor. %+v", err)
+		}
+		_, err = client.MonitoringV1().ServiceMonitors(namespace).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get servicemonitor. %+v", err)
+		}
+		sm, err = client.MonitoringV1().ServiceMonitors(namespace).Update(sm)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update servicemonitor. %+v", err)
+
+		}
+	}
+	return sm, err
+}
+
+// GetPrometheusRule returns a prometheusRule object
+func GetPrometheusRuleDefinition(name, namespace string) *monitoringv1.PrometheusRule {
+	prometheusRule := monitoringv1.PrometheusRule{}
+	prometheusRule.SetName(name)
+	prometheusRule.SetNamespace(namespace)
+	return &prometheusRule
+}
+
+// GetPrometheusRuleSpec returns a ruleSpec from provided rules or an error
+func GetPrometheusRuleSpec() (*monitoringv1.PrometheusRuleSpec, error) {
+	ruleFilePath := "/etc/ceph-monitoring/prometheus-ceph-rules.yaml"
```

Comment:
Can we make this path configurable? Or this is always going to be this path in operator pod?

Reply:
it is fixed. can't be made configurable.